packages/tinybased/src/lib/search/searched.ts

```diff
@@ -0,0 +1,101 @@
+/* eslint-disable @typescript-eslint/ban-types */
+import {
+  create,
+  getByID,
+  insert,
+  insertBatch,
+  Lyra,
+  remove,
+  search,
+  SearchParams,
+  SearchResult,
+} from '@lyrasearch/lyra';
+
+import { ObjectToCellStringType, TableBuilder } from '../TableBuilder';
+import { OnlyStringKeys, TinyBaseSchema } from '../types';
+
+type LyraInstances<TSchema extends TinyBaseSchema> = {
+  [K in OnlyStringKeys<TSchema>]: Lyra<ObjectToCellStringType<TSchema[K]>>;
+};
+export class Searched<TSchema extends TinyBaseSchema = {}> {
+  public static ERRORS = {
+    NO_TABLES_DEFINED: 'No tables defined for search',
+    TABLE_ALREADY_DEFINED: (table: string) => `Table ${table} already defined.`,
+    INDEX_NOT_CONFIGURED: (index: string) =>
+      `Index ${index} not configured for search`,
+    TABLE_NOT_IN_SCHEMA_BUILDER: (tableKeyToIndex: string) =>
+      `Table ${tableKeyToIndex} cannot be indexed. It is not defined for the SchemaBuilder.`,
+  };
+
+  public readonly tables: Map<OnlyStringKeys<TSchema>, TableBuilder<any, any>> =
```

Comment:
I don't actually think we benefit from having the Map enforce `OnlyStringKeys<TSchema>`. I switched it to `string` locally and it doesn't break any types but it allows us to drop a ton of `any` types inside the class